### PR TITLE
QML: Add support for dropping tracks on Decks

### PR DESCRIPTION
**Pull request by Holzhaus**
This allows you to open a file manager and drag-and-drop a file onto a deck. Tested with nautilus.

### CODE REVIEW

File: res/skins/QMLDemo/Deck.qml [hunk 1]
```diff
@@ -369,4 +370,15 @@ Item {
...
     }
 
+    DropArea {
+        anchors.fill: parent
+        onDropped: {
+            if (drop.hasUrls) {
+                let url = drop.urls[0];
+                console.log("Dropped URL '" + url + "' on deck " + root.group);
+                root.deckPlayer.loadTrackFromLocationUrl(url);
+            }
```
Comment (Swiftb0y) [inline res/skins/QMLDemo/Deck.qml:380]: idk if thats good etiquette, but I'd consider setting `drop.accepted = false` in an else case...

**Review by Swiftb0y — COMMENTED** [review]
I'd prefer is someone more familiar with C++ could take a look at that as well before merging.
There is a minor visual issue IMO: the window always indicates an "accepts drop" visual even when over no DropArea. Its slightly confusing, but I don't think we can do anything about that.

File: src/skin/qml/qmlplayermanagerproxy.cpp [hunk 2]
```diff
@@ -12,6 +12,10 @@ namespace qml {
 QmlPlayerManagerProxy::QmlPlayerManagerProxy(
         std::shared_ptr<PlayerManager> pPlayerManager, QObject* parent)
         : QObject(parent), m_pPlayerManager(pPlayerManager) {
+    connect(this,
+            &QmlPlayerManagerProxy::loadLocationToPlayer,
+            m_pPlayerManager.get(),
```
[the remark this answers is not in the document]
Comment (Holzhaus) [reply src/skin/qml/qmlplayermanagerproxy.cpp:17]: I don't think there is an issue. The player Manager outlives the skin anyway, and the QmlPlayerProxy object is managed by Qt.

[the remark this answers is not in the document]
Comment (uklotzde) [reply src/skin/qml/qmlplayermanagerproxy.cpp:17]: Since we also store the shared pointer as a member this should be safe.

**Review by uklotzde — APPROVED** [review]

**Review by Swiftb0y — APPROVED** [review]

**Pull request merged**